system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -2627,11 +2632,11 @@ class CLdapClient : public CInterface, implements ILdapClient
                   attribute != NULL;
                   attribute = atts.getNext())
             {
-                if(0 == stricmp(attribute, "cn"))
+                if(0 == stricmp(attribute, "distinguishedName"))
                 {
                     CLDAPGetValuesWrapper vals(ld, message, attribute);
                     if (vals.hasValues())
-                        userdn.append("cn=").append(vals.queryValues()[0]).append(",").append(m_ldapconfig->getUserBasedn());
+                        userdn.set(vals.queryValues()[0]);
```

Comment:
The following call to ldap_modify_s() requires a fully qualified DN distinguished name (ie ou=wwhitehead,ou=users,ou=ecl,DC=RISKDev,DC=LOCAL). In the legacy code, this was built up from the "usersBaseDN" setting in the config, followed by the "cn" that was queried before this change.  Because now we support nested users (ie they reside somewhere below the "users"), I need to know the fully qualified DN, including the levels below "users".  I tested this thoroughly in both legacy and RISKDev environments.